### PR TITLE
Don't explicitly commit the composing text in `_finalizeEditing`

### PR DESCRIPTION
**Pull request by LongCatIsLooong**
Related: https://github.com/flutter/flutter/issues/78827. Let IME keep its composing region. If `shouldUnfocus` is true the composing indicator will be removed when we close the connection. 

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/widgets/editable_text.dart [hunk 1]
```diff
@@ -1074,9 +1086,15 @@ class EditableText extends StatefulWidget {
...
-  /// next time the user inserts or deletes text.
+  /// Formatters are run in the provided order when the text input is changed by
+  /// the user, or the composing text is committed.
+  ///
+  /// Typically, text modification should only be applied when text is being
+  /// committed by the IME and not on text under composition (i.e., only when
+  /// [TextEditingValue.composing] is collapsed).
+  ///
+  /// When this parameter changes, the new formatters will not be applied until
+  /// the next time the user inserts or deletes text.
```
Comment (justinmc) [inline packages/flutter/lib/src/widgets/editable_text.dart:1097]: Does [MaxLengthEnforcement](https://github.com/flutter/flutter/pull/68086) affect this?

**Review by justinmc — APPROVED** [review]
This LGTM as long as the engine PR https://github.com/flutter/engine/pull/25180 is merged.

File: packages/flutter/lib/src/widgets/editable_text.dart (hunk 1 above)
Comment (LongCatIsLooong) [reply packages/flutter/lib/src/widgets/editable_text.dart:1097]: Are you referring to the second paragraph? Unfortunately some IMEs (shimeji, for instance) restore the composing text on the next keypress after we format its composing text. It should be fine now that we restart the IME on these occasions, but it comes with a performance cost so we don't want to do the restart too frequently, thus the recommendation.

**Comment by goderbauer** [comment]
(PR triage): Is this PR good to go?

**Comment by LongCatIsLooong** [comment]
Will get back to this later: the web failure seems to be a legitimate bug.

**Pull request closed without merging**